### PR TITLE
Ventura Permissions Issue

### PR DESCRIPTION
**Pull request by pierrehilbert**
## What does this PR do?

Monkey patch to fix the Ventura permission problem.
My PR needs to be reviewed by someone with more understanding about why it has been implemented that way (https://github.com/elastic/elastic-agent/commit/48a4703cf28ba7fd05d8f2cb1680438514a78736)

During my investigation time, I discovered that installing Elastic Agent and then Enroll it was working.
`sudo ./elastic-agent install -f`
`sudo ./elastic-agent enroll --url= --enrollment-token= `

The problem happens only when we are doing both in the same time
`sudo ./elastic-agent install --url= --enrollment-token=`

By investigating into the code, I discovered that during enrollment in the same time than installation, we are triggering two times the  FixPermissions() func

According to me, there is no reason to trigger this function during enrollment time and we should do it only when installing the binaries.

## Why is it important?

We can install or upgrade Elastic Agent on Mac OS Ventura.

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)

## Related issues

- Closes https://github.com/elastic/elastic-agent/issues/2103

### CODE REVIEW

File: internal/pkg/agent/cmd/enroll.go
Comment (cmacknz) [inline internal/pkg/agent/cmd/enroll.go:331]: How the agent is initialized differs between Fleet mode and Managed mode. An enrolment is essentially a switch between these two cases:

https://github.com/elastic/elastic-agent/blob/fd26c38120077a91ce81a0caeacf3b6ba7d0abce/internal/pkg/agent/application/application.go#L87-L125

The biggest difference I could spot that could be affected by the permissions of the agent is the creation of the encrypted configuration state store, which is triggered on this path:

https://github.com/elastic/elastic-agent/blob/fd26c38120077a91ce81a0caeacf3b6ba7d0abce/internal/pkg/agent/application/application.go#L101-L102

We would have to look at the permissions it ends up with if we remove this, and then also decide if it matters. The encryption is still there, but we would want to ensure it isn't readable by a non-root user.

Comment (pierrehilbert) [reply internal/pkg/agent/cmd/enroll.go:331]: From my local tests: 
`-rw-------   1 root  wheel     734 Feb 23 15:22 fleet.enc`

Comment (cmacknz) [reply internal/pkg/agent/cmd/enroll.go:331]: Great, those are created by the agent running as root so I hoped that would be the case. 

I think the logic to fix the permissions exists to change the ownership to root from the user that downloaded the installation package. This shouldn't apply to the enroll command.

I think this looks good, but let's wait for @blakerouse to take a look before we merge this.

We will also need to test this when the agent is signed, which likely requires merging this and testing a snapshot build. We need to go through the unified release process to get the binary signed. It shouldn't make a difference but we need to confirm.

Comment (cmacknz) [reply internal/pkg/agent/cmd/enroll.go:331]: The original change was from Blake here so hopefully he remembers if there was a reason we did this :) https://github.com/elastic/elastic-agent/commit/48a4703cf28ba7fd05d8f2cb1680438514a78736

**Comment by joshdover** [comment]
@blakerouse can we write an integration test for installs and enrolls that runs on each platform?

File: internal/pkg/agent/cmd/enroll.go
Comment (blakerouse) [reply internal/pkg/agent/cmd/enroll.go:331]: The original change was done to solve this issue - https://github.com/elastic/beats/issues/26213

I got a felling that changing this will result in it not working on Windows.

Comment (cmacknz) [reply internal/pkg/agent/cmd/enroll.go:331]: Since we only care about fixing this on Macs, the safest thing to do is to only set this to `false` when `runtime.GOOS == "darwin"` so it only takes effect on Mac.

Comment (pierrehilbert) [reply internal/pkg/agent/cmd/enroll.go]: I did the change, if someone can review it, I would appreciate!

File: internal/pkg/agent/cmd/enroll.go
Comment (cmacknz) [inline internal/pkg/agent/cmd/enroll.go:325]:
```suggestion
// On MacOS Ventura and above, fixing the permissions on enrollment during installation fails with the error:
//  Error: failed to fix permissions: chown /Library/Elastic/Agent/data/elastic-agent-c13f91/elastic-agent.app: operation not permitted
// This is because we are fixing permissions twice, once during installation and again during the enrollment step.
// When we are enrolling as part of installation on MacOS, skip the second attempt to fix permissions.
```

Let's add a more detailed explanation of exactly what the error is and why it happens. It will save whoever reads this next from having to go through the Git history and extremely long conversation in the linked issue.

**Comment by cmacknz** [comment]
You need to add a changelog fragment :)

**Comment by cmacknz** [comment]
Tested on my MacOS Monterey machine, installs + enrolls without issue. Let's fix up the comment and add a changelog entry and then I'll approve this.

File: changelog/fragments/1678553750-ventura-permission-issue.yaml
Comment (cmacknz) [inline changelog/fragments/1678553750-ventura-permission-issue.yaml:14]:
```suggestion
summary: Fix permission issue on MacOS Ventura and above when enrolling as part of the installation.
```

I don't think this affects Monterey and lower.

**Review by cmacknz — APPROVED** [review]

**Pull request merged**
